app/assets/javascripts/test/backend-snapshot-tests/annotations.e2e.js

```diff
@@ -0,0 +1,137 @@
+  t.is(reopenedAnnotation.state, "Active");
+  // $FlowFixMe: Make tracingTime deterministic
+  reopenedAnnotation.task.tracingTime = 100;
+  // reopenedAnnotation.tracingTime = 100;
+
+  t.snapshot(reopenedAnnotation, { id: "annotations-reOpenAnnotation" });
+});
+
+test.serial("finishAnnotation() and reOpenAnnotation() for explorational", async t => {
+  const annotationId = "68135c192faeb34c0081c05d";
+  const finishedAnnotation = await api.finishAnnotation(
+    annotationId,
+    APITracingTypeEnum.Explorational,
+  );
+  t.is(finishedAnnotation.state, "Finished");
+  // $FlowFixMe: Make tracingTime deterministic
+  finishedAnnotation.tracingTime = 100;
+  t.snapshot(finishedAnnotation, { id: "annotations-finishAnnotation-explorational" });
+
+  const reopenedAnnotation = await api.reOpenAnnotation(
+    annotationId,
+    APITracingTypeEnum.Explorational,
+  );
+  // $FlowFixMe: Make tracingTime deterministic
+  finishedAnnotation.tracingTime = 100;
+  t.is(reopenedAnnotation.state, "Active");
+
+  t.snapshot(reopenedAnnotation, { id: "annotations-reOpenAnnotation-explorational" });
+});
+
+test.serial("editAnnotation()", async t => {
+  const annotationId = "68135c192faeb34c0081c05d";
+  const originalAnnotation = await api.getAnnotationInformation(
+    annotationId,
+    APITracingTypeEnum.Explorational,
+  );
+  const { name, isPublic, description } = originalAnnotation;
+
+  const newName = "new name";
+  const newIsPublic = !isPublic;
+  const newDescription = "new description";
+
+  await api.editAnnotation(annotationId, APITracingTypeEnum.Explorational, {
+    name: "new name",
+    isPublic: newIsPublic,
+    description: newDescription,
+  });
+  const editedAnnotation = await api.getAnnotationInformation(
+    annotationId,
+    APITracingTypeEnum.Explorational,
+  );
+
+  t.is(editedAnnotation.name, newName);
+  t.is(editedAnnotation.isPublic, newIsPublic);
+  t.is(editedAnnotation.description, newDescription);
+  // $FlowFixMe: Make tracingTime deterministic
+  editedAnnotation.tracingTime = 100;
+  t.snapshot(editedAnnotation, { id: "annotations-editAnnotation" });
+
+  await api.editAnnotation(annotationId, APITracingTypeEnum.Explorational, {
+    name,
+    isPublic,
+    description,
+  });
+});
+
+test.serial("finishAllAnnotations()", async t => {
+  const annotationIds = ["78135c192faeb34c0081c05d", "78135c192faeb34c0081c05e"];
+
+  await api.finishAllAnnotations(annotationIds);
+
+  const finishedAnnotations = await Promise.all(
+    annotationIds.map(id => api.getAnnotationInformation(id, "skeleton")),
+  );
+
+  t.is(finishedAnnotations.length, 2);
+  finishedAnnotations.forEach(annotation => {
+    t.is(annotation.state, "Finished");
+  });
+
+  await Promise.all(annotationIds.map(id => api.reOpenAnnotation(id, APITracingTypeEnum.Task)));
+});
+
+// Tests which require a working dataStore during tests and therefore don't work yet:
```

Comment:
Do you know whether it would be hard to run a datastore during testing? Anyways, we should create a followup issue to do this and enable these tests :)

Reply:
I think, it should not be too hard. #2190 keeps track of it :)